sale_promotion_domain_product_discount/models/sale_order.py

```diff
@@ -0,0 +1,77 @@
+            )
+            for line in lines:
+                discount_line_amount = min(
+                    self._get_reward_values_discount_percentage_per_line(program, line),
+                    amount_total - currently_discounted_amount,
+                )
+                if discount_line_amount:
+                    if line.tax_id in reward_dict:
+                        reward_dict[line.tax_id]["price_unit"] -= discount_line_amount
+                    else:
+                        taxes = self.fiscal_position_id.map_tax(line.tax_id)
+                        uom_id = program.discount_line_product_id.uom_id.id
+                        reward_dict[line.tax_id] = {
+                            "name": _(
+                                "Discount: %(program)s - On product with following "
+                                "taxes: %(taxes)s",
+                                program=program.name,
+                                taxes=", ".join(taxes.mapped("name")),
+                            ),
+                            "product_id": program.discount_line_product_id.id,
+                            "price_unit": -discount_line_amount
+                            if discount_line_amount > 0
+                            else 0,
+                            "product_uom_qty": 1.0,
+                            "product_uom": uom_id,
+                            "is_reward_line": True,
+                            "tax_id": [(4, tax.id, False) for tax in taxes],
+                        }
+                        currently_discounted_amount += discount_line_amount
+
+            # If there is a max amount for discount, we might have to limit
+            # some discount lines or completely remove some lines
+            max_amount = program._compute_program_amount(
+                "discount_max_amount", self.currency_id
+            )
+            if max_amount > 0:
+                amount_already_given = 0
+                for val in list(reward_dict):
+                    amount_to_discount = (
+                        amount_already_given + reward_dict[val]["price_unit"]
+                    )
+                    if abs(amount_to_discount) > max_amount:
+                        reward_dict[val]["price_unit"] = -(
+                            max_amount - abs(amount_already_given)
+                        )
+                        add_name = formatLang(
+                            self.env, max_amount, currency_obj=self.currency_id
+                        )
+                        reward_dict[val]["name"] += (
+                            "( " + _("limited to ") + add_name + ")"
+                        )
+                    amount_already_given += reward_dict[val]["price_unit"]
+                    if reward_dict[val]["price_unit"] == 0:
+                        del reward_dict[val]
+            return reward_dict.values()
+        else:
+            return super()._get_reward_values_discount(program)
```

Comment:
```suggestion
        if program.discount_apply_on != "domain_product":
            return super()._get_reward_values_discount(program)
```
Do this on top and you save indentation for the whole method.